schematic/manifest/generator.py

```diff
@@ -1655,8 +1639,27 @@ def create_manifests(
             use_annotations (bool, optional): whether to use annotations. Defaults to False.
 
         Returns:
-            Union[List[str], List[pd.DataFrame], BinaryIO]: a list of Googlesheet URLs, a list of pandas dataframes or an Excel file.
+            Union[List[str], List[pd.DataFrame]]: a list of Googlesheet URLs, a list of pandas dataframes or excel file paths
         """
+        if dataset_ids:
+            # Check that the number of submitted data_types matches
+            # the number of dataset_ids (if applicable)
+            len_data_types = len(data_types)
+            len_dataset_ids = len(dataset_ids)
+
+            if len_data_types != len_dataset_ids:
+                raise ValueError(
+                    f"There is a mismatch in the number of data_types and dataset_id's that "
+                    f"submitted. Please check your submission and try again."
+                )
+
+            # Raise an error if used in conjunction with datatype = 'all_manifests'
+            if data_types[0] == "all manifests":
+                raise ValueError(
+                    f"When submitting 'all manifests' as the data_type cannot also submit dataset_id. "
+                    f"Please check your submission and try again."
+                )
```

Comment:
Whats the reason for using f-strings here?

Reply:
This is legacy code. Let me change the f strings here.